v-next/hardhat/src/internal/core/config-validation.ts

```diff
@@ -613,20 +629,39 @@ export function validatePluginsConfig(
     }
 
     if (plugin.dependencies !== undefined) {
-      if (Array.isArray(plugin.dependencies)) {
-        for (const [depIndex, dep] of plugin.dependencies.entries()) {
-          if (typeof dep !== "function") {
-            validationErrors.push({
-              path: [...path, "plugins", index, "dependencies", depIndex],
-              message: "plugin dependencies must be an array of functions",
-            });
-          }
-        }
-      } else {
+      if (typeof plugin.dependencies !== "function") {
         validationErrors.push({
           path: [...path, "plugins", index, "dependencies"],
-          message: "plugin dependencies must be an array",
+          message: "plugin dependencies must be a function returning an array",
         });
+      } else {
+        let result: unknown;
+
+        try {
+          result = plugin.dependencies();
```

Comment:
This also goes away, because otherwise we'll be loading the dependencies all the time. Same for actions and hook handlers. All we can validate is that they are functions.